stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -762,6 +762,29 @@ function eigen(D::Diagonal; permute::Bool=true, scale::Bool=true, sortby::Union{
     end
     Eigen(λ, evecs)
 end
+function eigen(Da::Diagonal, Db::Diagonal; sortby::Union{Function,Nothing}=nothing)
+    if any(!isfinite, Da.diag) || any(!isfinite, Db.diag)
+        throw(ArgumentError("matrices contain Infs or NaNs"))
+    end
+    if !isposdef(Db)
+        throw(ArgumentError("right-hand side diagonal matrix is not positive definit"))
```

Comment:
definite (several occurences)

Reply:
don't errors have dots at the end?

Reply:
Most errors don't have a dot at the end, and if is has, we should probably start with a capital letter. You're right about the invertibility. That's what I had first, actually, so I'll revert to `any(iszero, D.diag)`.